src/Core/src/Platform/MauiContext.cs

```diff
@@ -4,7 +4,27 @@
 
 namespace Microsoft.Maui
 {
-	public partial class MauiContext : IMauiContext, IScopedMauiContext
+	public partial class MauiApplicationContext : IMauiApplicationContext
+	{
+		public MauiApplicationContext()
+		{
+			// Temporary hack until we fully remove Forms.Init
+			Services = null!;
+			Handlers = null!;
+		}
+
+		public MauiApplicationContext(IServiceProvider services)
+		{
+			Services = services ?? throw new ArgumentNullException(nameof(services));
+			Handlers = Services.GetRequiredService<IMauiHandlersServiceProvider>();
+		}
+
+		public IServiceProvider Services { get; }
+
+		public IMauiHandlersServiceProvider Handlers { get; }
+	}
+
+	public partial class MauiContext : IMauiWindowContext, IScopedMauiContext
```

Comment:
The original `MauiContext` is now a `IMauiWindowContext` (which is `IMauiContext` still).

Reply:
Should this MauiContext have a reference to MauiApplicationContext?

And then we could have an extension method

GetApplicationContext(this IMauiContext context)

That makes it easy to roll up through the MauiContext to get to the application context?